src/Tools/ExternalAccess/Razor/Cohost/RazorDynamicRegistrationServiceFactory.cs

```diff
@@ -66,23 +67,25 @@ public Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestCon
 
             void InitializeRazor()
             {
-                this.InitializeRazor(clientCapabilities, context, _disposalTokenSource.Token);
+                this.InitializeRazorAsync(clientCapabilities, context, _disposalTokenSource.Token).ReportNonFatalErrorAsync();
             }
         }
 
-        private void InitializeRazor(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
+        private async Task InitializeRazorAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
         {
             // The LSP server will dispose us when the server exits, but VS could decide to activate us later.
             // If a new instance of the server is created, a new instance of this class will be created and the
             // UIContext will already be active, so this method will be immediately called on the new instance.
             if (cancellationToken.IsCancellationRequested) return;
 
+            await TaskScheduler.Default.SwitchTo(alwaysYield: true);
```

Comment:
>SwitchTo(alwaysYield: true); [](http://example.com/codeflow?start=40&length=28)

Just so I'm clear, the SwitchTo isn't necessary as the concern is about leaving the main thread, right?

Reply:
Oh, or maybe you just don't want to block the calling thread, even if it's a threadpool thread. I see.

Reply:
Probably not necessary in the real world, but in their the UI context could have been activated before we got here (in future if the conditions for the context change), or we could be in VS Code (in future) and I thought getting out of the way of the LSP servers initialize handler is probably kind regardless.